### ISSUE TITLE
Errors with "around" selection command in version 0.16

### ISSUE THREAD

**Issue by maxscheurer** [MDAnalysis/mdanalysis#1311]
### Expected behaviour
Correct number of atoms in selection with "around" keyword.

### Actual behaviour
As of version 0.16.0, the atom selection command "around" does not yield the same
results as version 0.15 .
The results gained from the older version are in agreement with VMD's "within" selection keyword, i.e.
``` tcl
set sel [atomselect top "segname A and within 5 of segname B"]
```
Perhaps I missed something important in the Release Notes that explains this behavior...

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda

u = mda.Universe(top, trj)
u.select_atoms("segid A and around 5 segid B")

```

### Currently version of MDAnalysis:
0.16.0

**Comment by richardjgowers** [comment]
This is very possible, I did refactor the around selection. Can you make an
example from our test files Max?

**Comment by maxscheurer** [comment]
I will try to do so asap...

**Comment by richardjgowers** [comment]
@maxscheurer if it's tricky to make it not work using the test files you can also just upload a system here where you're getting different results

**Comment by orbeckst** [comment]
Thanks for reporting. An example would be crucial, though.

**Comment by orbeckst** [comment]
@maxscheurer have you had time to look at an example?

**Comment by orbeckst** [comment]
ping @maxscheurer ... we would like to look at this issue but we really need something to work with.

**Comment by maxscheurer** [comment]
@orbeckst, I will try to upload something tomorrow. Thanks for the reminder!

**Comment by maxscheurer** [comment]
@orbeckst, the issue seems to be fixed in MDAnalysis version 0.16.1
I tested with VMD; mda version 0.15; 0.16.0 and 0.16.1 
and the latter one yields the correct result again (as VMD and mda 0.15).

**Comment by richardjgowers** [comment]
Just in case this crops up later, I think what happened here wasn't actually the `around` selection itself, but rather the `segid A` selections working incorrectly in 0.16.  In 0.16.1 we improved/fixed how segments were defined in a few cases.
